### DISCUSSION TITLE
Will CSharp based React Native windows project be supported going forward with the "new architecture"?

### DISCUSSION THREAD

**Discussion by IamJignesh** [microsoft/react-native-windows#13628] [Q&A] [answered]
We are starting a new React Native windows project and are looking for recommendation on choosing between c# or c++ for the windows part of the app.

Initially we were using this documentation https://github.com/microsoft/react-native-windows/tree/main/packages/react-native-windows-init where we could specify the language as c# or c++ but now it seems that the documentation has been updated and the c# options has been labeled as "old/uwp-cs-app" : here https://microsoft.github.io/react-native-windows/docs/init-windows-cli
 "old/uwp-cs-app" | React Native Windows Application (Old Arch, UWP, C#) 

We are more inclined towards c# as we have more experience with it as well as we have a lot of c# code that we would like to reuse but we are not sure if it is the right choice going forward with the "new architecture".

**Comment by jonthysell** [comment, marked as answer]
Right now, the (experimental) new architecture templates are C++ as we work out the whole E2E of apps and custom community modules.

**Reply by Mohammad-Reihani** [reply]
Any new updates? what now?

**Reply by ninjaboy667** [reply]
@jonthysell 
would you be willing to follow up on this? Will the new architecture only be supported in cpp? There is no template for the new architecture. Going forward will microsoft only support c++ for future versions of react-native for windows?

**Comment by wuleninja** [comment]
Also very interested in knowing if I should invest time in C# native modules, since the current documentation seems rather shifted towards C++. I'd like for C# native modules to work since the DX feels a lot better in this language, but I could not make an example work so far.

**Comment by nick-meske-ulteig** [comment]
I would also like to know if there is a roadmap to get support for C# on the new architecture. We used the old template for C# but the required version of .net is significantly hindering for what we want to do.

**Comment by Katerlad** [comment]
Hello! @jonthysell, I was hoping you may have some insider info on how far along it may be until we see some C# version of the new architecture, are we looking at this year? or further?
